core/setup.py

```diff
@@ -45,18 +47,23 @@
     ]
 )
 
-extras_require = {
-    "ray": [
-        "ray[default]>=2.6.3,<2.7",
-        "async-timeout",
-    ],
-    "raytune": [
-        "ray[default,tune]>=2.6.3,<2.7",
-        # TODO: consider alternatives as hyperopt is not actively maintained.
-        "hyperopt>=0.2.7,<0.2.8",  # This is needed for the bayes search to work.
-        # 'GPy>=1.10.0,<1.11.0'  # TODO: Enable this once PBT/PB2 are supported by ray lightning
-    ],
-}
+# Check for Windows and Python 3.11
+is_windows_and_py311 = platform.system().lower() == "windows" and sys.version_info.major == 3 and sys.version_info.minor == 11
```

Comment:
I don't think this will solve the problem when doing a pip install. The `setup.py` to my understanding is not executed in pip install.

Reply:
changed to environment markers and tested with `pip install autogluon-tonyhu-test==1.0.5b20240302`

Reply:
Is “win32” the correct identifier? What are the other “winXYZ” markers? Is there a list somewhere?